server/images.go

```diff
@@ -81,19 +100,58 @@ func (m *Model) Prompt(p PromptVars) (string, error) {
 		return "", err
 	}
 	prompt.WriteString(sb.String())
-	prompt.WriteString(p.Response)
+
+	if !strings.Contains(prompt.String(), p.Response) {
```

Comment:
Why doesn't this use `PreResponsePrompt` and `PostResponsePrompt` to write the full prompt?

Reply:
this is applying the variable to a given template, it will either get called with the pre or post responses, and not at the same time.

Reply:
In that case, shouldn't this be handled by either `PreResponsePrompt` or `PostResponsePrompt`? It's unclear what the relationship is between these three functions.

Reply:
This is the "apply all variables to a template" function. The pre/post prompts just handle trimming the template itself.

I drew a fancy picture to show my mental map.
<img width="1473" alt="image" src="https://github.com/jmorganca/ollama/assets/5853428/a1432713-6494-4b9d-921b-42d3b8dab19e">

Reply:
This feels weird to me. My mental model would've preprocessed and split the pre and post response templates and `*ResponsePrompt` will simply call `template.Execute`. That being said, this is fine too